cpp/pipelines/add_authority_external_ref.cc

```diff
@@ -18,120 +19,201 @@
  *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <chrono>
 #include <fstream>
 #include <iostream>
 #include <regex>
+#include <string>
+#include <nlohmann/json.hpp>
 #include "BeaconFile.h"
+#include "FileUtil.h"
 #include "MARC.h"
 #include "StringUtil.h"
+#include "TextUtil.h"
 #include "util.h"
 
-// Scenario 1:
-// use --create_mapping_file parameter <filepath> to generate the mapping file
-//  of the downloaded dnb authoriy dump (must bei unzipped first)
-//  Download from: https://data.dnb.de/opendata/authorities-gnd-person_lds.jsonld.gz
-//  and unzip to e.g. authorities-gnd-person_lds_20210613.jsonld
-//  output is stdout
-// Scenario 2:
-// use converted file from scenario 1 from cpp/data to create a map during
-//  pipeline processing. The norm_data_input is extended by wikidata ids where possible
-//  and saved to 024 field indicator1:7 where wikidata id is not yet present
-//  file can be taken from /mnt/ZE020150/FID-Entwicklung/ub_tools (gnd_to_wiki.txt)
+/**
+ * There are two scenarios to fullfil the task. The second scenario depend on the file generated by the first scenario.
+ *
+ * Scenario 1:
+ * use --create_mapping_file <input_txt_file> <output_csv_file> to generate the mapping file.
+ * All processes from downloading data to generating output file is managed by the script 'add_authority_external_ref.automation.py'
+ *
+ * Generating the input file for scenario 1 must use the jq program. Since the .jsonld file is large enough than it is a must to parse it as
```

Comment:
If i understand correctly, the jq-related handling will be done in the python program, so maybe the documentation here could also mention this (it is not necessary to do this by hand).